src/devices/machine/z80dma.cpp

```diff
@@ -670,7 +673,7 @@ void z80dma_device::write(u8 data)
 			if (data & 0x10)
 				m_regs_follow[m_num_follow++] = GET_REGNUM(MATCH_BYTE);
 
-			if (BIT(data, 6))
+			if (BIT(data, 6) && m_dma_mode != dma_mode::UA858D)
```

Comment:
Still not quite convinced that the base `z80dma_device` has to know it has children here, perhaps moving this logic to `ua858d_device` and make `specnext_dma_device` a subclass of that should be the play here?

Reply:
> Still not quite convinced that the base `z80dma_device` has to know it has children here, perhaps moving this logic to `ua858d_device` and make `specnext_dma_device` a subclass of that should be the play here?

I did this on purpose for Next case. It needs to switch type preserving existing state of registers.

Reply:
I mean, in the context of _if_ somebody else went their own route in subclassing the arch. 
Notice that even a `// NOTE:` or a `// FIXME:` is okay for me.

Reply:
Ok. Let's assume we keep specnextdma how it was before this commit.
Regarding spectrum which select chip based on mashine config... do I need to add slot for dma now? Or are the other way to do this?